java/core/src/java/org/apache/orc/impl/InStream.java

```diff
@@ -471,6 +472,7 @@ private int readHeaderByte() {
     }
 
     private void readHeader() throws IOException {
+      currentCompressedStart = compressed.position();
```

Comment:
I think this needs to be this.position.

Reply:
This is a good catch @omalley fixed this and added a test to verify that the seek skip takes place even when the DiskRange does not perfectly align with the stream